storehaus-cassandra/src/main/scala/com/twitter/storehaus/cassandra/cql/AbstractCQLCassandraCompositeStore.scala

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2014 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twitter.storehaus.cassandra.cql
+
+import com.twitter.util.{Duration, Future, FuturePool, Return, Time, Throw}
+import com.twitter.concurrent.Spool
+import com.datastax.driver.core.{BatchStatement, ConsistencyLevel, ResultSet, Row, Statement}
+import com.datastax.driver.core.policies.{LoadBalancingPolicy, Policies, ReconnectionPolicy, RetryPolicy, RoundRobinPolicy, TokenAwarePolicy}
+import com.datastax.driver.core.querybuilder.{BuiltStatement, Clause, QueryBuilder, Update, Select}
+import com.twitter.storehaus.{IterableStore, QueryableStore, ReadableStore, Store}
+import com.websudos.phantom.CassandraPrimitive
+import java.util.concurrent.{Executors, TimeUnit}
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ArrayBuffer
+import scala.language.implicitConversions
+import shapeless._
+import HList._
+import ops.hlist.Mapper
+import ops.hlist.Mapped
+import poly._
+import Nat._
+import UnaryTCConstraint._
+import org.slf4j.LoggerFactory
+
+object AbstractCQLCassandraCompositeStore {
+  
+  /**
+   * used for mapping the keys in the HLists to Strings
+   */
+  object keyStringMapping extends (CassandraPrimitive ~>> String) {
```

Comment:
I'd be very tempted to define a custom type class that provides a conversion directly from an `HList` of `CassandraPrimitive`s to a `List[String]`. It would save a lot of `Mapper` / `ToList` pairs below, and I personally think I'd find it clearer.